realm/realm-library/src/main/java/io/realm/RealmSchema.java

```diff
@@ -221,20 +221,38 @@ Table getTable(String className) {
     Table getTable(Class<? extends RealmModel> clazz) {
         Table table = classToTable.get(clazz);
         if (table == null) {
-            clazz = Util.getOriginalModelClass(clazz);
-            table = transaction.getTable(realm.configuration.getSchemaMediator().getTableName(clazz));
-            classToTable.put(clazz, table);
+            Class<? extends RealmModel> originalClass = Util.getOriginalModelClass(clazz);
+            if (originalClass != clazz) {
+                table = classToTable.get(originalClass);
+            }
+            if (table == null) {
+                table = transaction.getTable(realm.configuration.getSchemaMediator().getTableName(originalClass));
+                classToTable.put(originalClass, table);
+            }
+            if (originalClass != clazz) {
```

Comment:
No. `originalClass` is always model class which is defined by the user.
But `clazz` can be both user defined model class (`originalClass == class`) and generated proxy class (`originalClass != class`).
In latter case, we need to create another cache entry.